pyls/hookspecs.py

```diff
@@ -23,7 +23,12 @@ def pyls_commands(config, workspace):
 
 
 @hookspec
-def pyls_completions(config, workspace, document, position):
+def pyls_rope_completions(config, workspace, document, position):
```

Comment:
We should keep `pyls_completions` and just have the rope and jedi plugins implement them separately.

Reply:
Well, by doing this, you cannot control each implementation separately, instead both are called simultaneously

Reply:
That's what I meant by moving the race logic into the pluggy layer. So for any pluggy hook we can race the results. I can take a look to see if its possible